src/server/object_services/object_server.js

```diff
@@ -406,9 +406,14 @@ async function complete_object_upload(req) {
         set_updates.sha256_b64 = req.rpc_params.sha256_b64;
     }
 
-    const map_res = req.rpc_params.multiparts ?
-        await _complete_object_multiparts(obj, req.rpc_params.multiparts) :
-        await _complete_object_parts(obj);
+    let map_res;
+    if (req.rpc_params.multiparts) {
+        map_res = await _complete_object_multiparts(obj, req.rpc_params.multiparts);
+    } else if (req.rpc_params.num_parts) {
+        map_res = {size: obj.size, num_parts: req.rpc_params.num_parts};
```

Comment:
add short explanation
```suggestion
    } else if (req.rpc_params.num_parts) {
        // in this case we got all the info we need and can avoid quering md_store
        map_res = { size: obj.size, num_parts: req.rpc_params.num_parts };
```